modules/memfault/Kconfig

```diff
@@ -0,0 +1,133 @@
+	string "Firmware type"
+	default "nrf91ns-fw" if BOARD_NRF9160DK_NRF9160NS || BOARD_THINGY91_NRF9160NS
+	help
+	  Firmware type running on the board.
+
+choice
+	prompt "Firmware version generation method"
+	default MEMFAULT_NCS_FW_VERSION_AUTO
+
+config MEMFAULT_NCS_FW_VERSION_STATIC
+	bool "Static firmware version"
+	help
+	  Use statically configured firmware version, as opposed to an
+	  dynamically generated one.
+
+config MEMFAULT_NCS_FW_VERSION_AUTO
+	bool "Dynamic firmware version"
+	help
+	  Use dyanimically generated firmware version. The version is
+	  re-generated on every compiler run.
+endchoice
+
+config MEMFAULT_NCS_FW_VERSION_PREFIX
+	string "Firmware version prefix"
+	depends on MEMFAULT_NCS_FW_VERSION_AUTO
+	default "0.0.1+"
+	help
+	  Prefix to use in front of automatically generated build version string.
+	  If the build version string is "a1b2c3d", the full firmware version
+	  will become "<prefix>a1b2c3d".
+
+config MEMFAULT_NCS_INIT_PRIORITY
+	int "memfault initialization priority"
+	range AT_CMD_INIT_PRIORITY 99 if MEMFAULT_NCS_DEVICE_SERIAL_USE_IMEI
+	default 90
+	help
+	  If IMEI is used as device ID, the initialization priority must be
+	  greater than that of the AT command module.
+
+config MEMFAULT_NCS_PROVISION_CERTIFICATES
+	bool "Provision certificates on initalization"
+	depends on  MEMFAULT_ROOT_CERT_STORAGE_NRF9160_MODEM
+	default y
+	help
+	  Provision required TLS root certificates for connecting to Memfault's
+	  endpoints. If this option is disabled, the user is responsible for
+	  calling the appropriate Memfault APIs for provisioning the
+	  certificates before attempting to connect to Memfault servers.
+
+config MEMFAULT_NCS_STACK_METRICS
+	bool "Collect stack metrics"
+	select THREAD_ANALYZER
+	select THREAD_NAME
+	default y
+	help
+	  Collect metrics for unused stack space for selected stacks.
+	  Currently the following stacks' unused space is monitored:
+		- at_cmd_socket_thread, used by the AT command library
+		- connection_poll_thread, used by the cloud libraries for
+		  nRF Cloud, AWS IoT and Azure IoT Hub
+
+config MEMFAULT_NCS_STACK_CHECK_INTERVAL
+	int "Stack check interval in seconds"
+	depends on MEMFAULT_NCS_STACK_METRICS
+	default 3600
+	help
+	  Stack checks are performed periodically. This configuration option
+	  can be used to change how often the stack check is performed.
+
+config MEMFAULT_NCS_LTE_METRICS
+	bool "Collect LTE metrics"
+	select LTE_LC_TRACE
+	depends on LTE_LINK_CONTROL
+	help
+	  Capture LTE metrics while the application is running. Supported
+	  metrics are time to connect and number of connection establishments.
+
+config MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP
+	bool "Save coredump to internal flash instead of RAM"
+	select MEMFAULT_COREDUMP_STORAGE_CUSTOM
+	depends on FLASH
+	depends on IMG_MANAGER
+	depends on BOOTLOADER_MCUBOOT
+	depends on MCUBOOT_IMG_MANAGER
+
+if MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP
+partition=MEMFAULT_STORAGE
```

Comment:
Is there a reason not to invoke the template as is done here?
https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/partition_manager/Kconfig#L40

Reply:
I initially tried to have this in the Memfault module's Kconfig, but I was not able to get it to work. Forgot to restore the template code when I moved it back here.
Is it supposed to work to have this code in another Kconfig file? The issue was that none of the partition macros were present in the generated `pm_config.h`, so compilation failed